--- client.mk.orig	2006-03-06 20:55:51.000000000 +0100
+++ client.mk	2006-10-14 13:00:02.000000000 +0200
@@ -107,17 +107,17 @@
 
 MODULES_core :=                                 \
   SeaMonkeyAll                                  \
-  mozilla/browser/config/version.txt            \
-  mozilla/mail/config/version.txt               \
-  mozilla/ipc/ipcd                              \
-  mozilla/modules/libpr0n                       \
-  mozilla/modules/libmar                        \
-  mozilla/modules/libbz2                        \
-  mozilla/accessible                            \
-  mozilla/security/manager                      \
-  mozilla/toolkit                               \
-  mozilla/storage                               \
-  mozilla/db/sqlite3                            \
+  browser/config/version.txt            \
+  mail/config/version.txt               \
+  ipc/ipcd                              \
+  modules/libpr0n                       \
+  modules/libmar                        \
+  modules/libbz2                        \
+  accessible                            \
+  security/manager                      \
+  toolkit                               \
+  storage                               \
+  db/sqlite3                            \
   $(NULL)
 
 LOCALES_core :=                                 \
@@ -127,7 +127,7 @@
 
 MODULES_toolkit :=                              \
   $(MODULES_core)                               \
-  mozilla/chrome                                \
+  chrome                                \
   $(NULL)
 
 LOCALES_toolkit :=                              \
@@ -138,7 +138,7 @@
 
 MODULES_suite :=                                \
   $(MODULES_core)                               \
-  mozilla/suite                                 \
+  suite                                 \
   $(NULL)
 
 LOCALES_suite :=                                \
@@ -147,9 +147,9 @@
 
 MODULES_browser :=                              \
   $(MODULES_toolkit)                            \
-  mozilla/browser                               \
-  mozilla/other-licenses/branding/firefox       \
-  mozilla/other-licenses/7zstub/firefox         \
+  browser                               \
+  other-licenses/branding/firefox       \
+  other-licenses/7zstub/firefox         \
   $(NULL)
 
 LOCALES_browser :=                              \
@@ -158,18 +158,18 @@
   other-licenses/branding/firefox               \
   $(NULL)
 
-BOOTSTRAP_browser := mozilla/browser/config/mozconfig
+BOOTSTRAP_browser := browser/config/mozconfig
 
 MODULES_minimo :=                               \
   $(MODULES_toolkit)                            \
-  mozilla/minimo                                \
+  minimo                                \
   $(NULL)
 
 MODULES_mail :=                                 \
   $(MODULES_toolkit)                            \
-  mozilla/mail                                  \
-  mozilla/other-licenses/branding/thunderbird   \
-  mozilla/other-licenses/7zstub/thunderbird     \
+  mail                                  \
+  other-licenses/branding/thunderbird   \
+  other-licenses/7zstub/thunderbird     \
   $(NULL)
 
 LOCALES_mail :=                                 \
@@ -179,49 +179,49 @@
   editor/ui                                     \
   $(NULL)
 
-BOOTSTRAP_mail := mozilla/mail/config/mozconfig
+BOOTSTRAP_mail := mail/config/mozconfig
 
 MODULES_composer :=                             \
   $(MODULES_toolkit)                            \
-  mozilla/composer                              \
+  composer                              \
   $(NULL)
 
 MODULES_calendar :=                             \
   $(MODULES_toolkit)                            \
-  mozilla/storage                               \
-  mozilla/db/sqlite3                            \
-  mozilla/calendar                              \
+  storage                               \
+  db/sqlite3                            \
+  calendar                              \
   $(NULL)
 
-BOOTSTRAP_calendar := mozilla/calendar/sunbird/config/mozconfig
+BOOTSTRAP_calendar := calendar/sunbird/config/mozconfig
 
 MODULES_xulrunner :=                            \
   $(MODULES_toolkit)                            \
-  mozilla/xulrunner                             \
+  xulrunner                             \
   $(NULL)
 
 LOCALES_xulrunner :=                            \
   $(LOCALES_toolkit)                            \
   $(NULL)
 
-BOOTSTRAP_xulrunner := mozilla/xulrunner/config/mozconfig
+BOOTSTRAP_xulrunner := xulrunner/config/mozconfig
 
 MODULES_macbrowser :=                           \
   $(MODULES_core)                               \
-  mozilla/camino                                \
+  camino                                \
   $(NULL)
 
-BOOTSTRAP_macbrowser := mozilla/camino/config/mozconfig
+BOOTSTRAP_macbrowser := camino/config/mozconfig
 
 MODULES_all :=                                  \
-  mozilla/other-licenses/bsdiff                 \
-  mozilla/other-licenses/libart_lgpl            \
-  mozilla/tools/trace-malloc                    \
-  mozilla/tools/jprof                           \
-  mozilla/tools/codesighs                       \
-  mozilla/tools/update-packaging                \
-  mozilla/other-licenses/branding               \
-  mozilla/other-licenses/7zstub                 \
+  other-licenses/bsdiff                 \
+  other-licenses/libart_lgpl            \
+  tools/trace-malloc                    \
+  tools/jprof                           \
+  tools/codesighs                       \
+  tools/update-packaging                \
+  other-licenses/branding               \
+  other-licenses/7zstub                 \
   $(NULL)
 
 #######################################################################
@@ -256,7 +256,7 @@
 else
 # Ran from mozilla/.. directory (?)
 ROOTDIR   := $(CWD)
-TOPSRCDIR := $(CWD)/mozilla
+TOPSRCDIR := $(CWD)
 endif
 
 # on os2, TOPSRCDIR may have two forward slashes in a row, which doesn't
@@ -316,16 +316,16 @@
 
 # See build pages, http://www.mozilla.org/build/ for how to set up mozconfig.
 
-MOZCONFIG_LOADER := mozilla/build/autoconf/mozconfig2client-mk
-MOZCONFIG_FINDER := mozilla/build/autoconf/mozconfig-find 
-MOZCONFIG_MODULES := mozilla/build/unix/modules.mk mozilla/build/unix/uniq.pl
+MOZCONFIG_LOADER := build/autoconf/mozconfig2client-mk
+MOZCONFIG_FINDER := build/autoconf/mozconfig-find 
+MOZCONFIG_MODULES := build/unix/modules.mk build/unix/uniq.pl
 run_for_side_effects := \
   $(shell cd $(ROOTDIR); \
      if test "$(_IS_FIRST_CHECKOUT)"; then \
         $(CVSCO) $(MOZCONFIG_FINDER) $(MOZCONFIG_LOADER) $(MOZCONFIG_MODULES); \
      else true; \
      fi; \
-     $(MOZCONFIG_LOADER) $(TOPSRCDIR) mozilla/.mozconfig.mk > mozilla/.mozconfig.out)
+     $(MOZCONFIG_LOADER) $(TOPSRCDIR) .mozconfig.mk > .mozconfig.out)
 include $(TOPSRCDIR)/.mozconfig.mk
 include $(TOPSRCDIR)/build/unix/modules.mk
 
@@ -376,8 +376,8 @@
 # CVS defines for NSS
 #
 NSS_CO_MODULE =               \
-		mozilla/security/nss      \
-		mozilla/security/coreconf \
+		security/nss      \
+		security/coreconf \
 		$(NULL)
 
 NSS_CO_FLAGS := -P
@@ -396,7 +396,7 @@
 ####################################
 # CVS defines for NSPR
 #
-NSPR_CO_MODULE = mozilla/nsprpub
+NSPR_CO_MODULE = nsprpub
 NSPR_CO_FLAGS := -P
 ifdef MOZ_CO_FLAGS
   NSPR_CO_FLAGS := $(MOZ_CO_FLAGS)
@@ -413,7 +413,7 @@
 ####################################
 # CVS defines for the C LDAP SDK
 #
-LDAPCSDK_CO_MODULE = mozilla/directory/c-sdk
+LDAPCSDK_CO_MODULE = directory/c-sdk
 LDAPCSDK_CO_FLAGS := -P
 ifdef MOZ_CO_FLAGS
   LDAPCSDK_CO_FLAGS := $(MOZ_CO_FLAGS)
@@ -432,8 +432,8 @@
   STANDALONE_CO_MODULE += allmakefiles.sh client.mk aclocal.m4 configure configure.in
   STANDALONE_CO_MODULE += Makefile.in
 
-	MOZ_MODULE_LIST += $(addprefix mozilla/,$(STANDALONE_CO_MODULE))
-  NOSUBDIRS_MODULE := $(addprefix mozilla/,$(BUILD_MODULE_CVS_NS))
+	MOZ_MODULE_LIST += $(addprefix ,$(STANDALONE_CO_MODULE))
+  NOSUBDIRS_MODULE := $(addprefix ,$(BUILD_MODULE_CVS_NS))
 
 ifeq (,$(filter $(NSPRPUB_DIR), $(BUILD_MODULE_CVS))$(MOZ_CO_PROJECT))
   CVSCO_NSPR :=
@@ -451,11 +451,11 @@
 #
 
 ifdef MOZ_MAPINFO
-$(warning MOZ_MAPINFO is obsolete, use MOZ_CO_MODULE=mozilla/tools/codesighs instead.)
-MOZ_MODULE_LIST += mozilla/tools/codesighs
+$(warning MOZ_MAPINFO is obsolete, use MOZ_CO_MODULE=tools/codesighs instead.)
+MOZ_MODULE_LIST += tools/codesighs
 endif
 ifdef MOZ_INTERNAL_LIBART_LGPL
-$(error MOZ_INTERNAL_LIBART_LGPL is obsolete, use MOZ_CO_MODULE=mozilla/other-licenses/libart_lgpl instead.)
+$(error MOZ_INTERNAL_LIBART_LGPL is obsolete, use MOZ_CO_MODULE=other-licenses/libart_lgpl instead.)
 endif
 ifdef MOZ_PHOENIX
 $(warning MOZ_PHOENIX is obsolete.)
@@ -519,9 +519,9 @@
 override MOZ_CO_LOCALES := $(subst $(comma), ,$(MOZ_CO_LOCALES))
 
 ifeq (all,$(MOZ_CO_LOCALES))
-MOZCONFIG_MODULES += $(foreach project,$(MOZ_PROJECT_LIST),mozilla/$(project)/locales/all-locales)
+MOZCONFIG_MODULES += $(foreach project,$(MOZ_PROJECT_LIST),$(project)/locales/all-locales)
 
-LOCALE_CO_DIRS := $(sort $(foreach project,$(MOZ_PROJECT_LIST),$(foreach locale,$(shell cat mozilla/$(project)/locales/all-locales),$(foreach dir,$(LOCALES_$(project)),l10n/$(locale)/$(dir)))))
+LOCALE_CO_DIRS := $(sort $(foreach project,$(MOZ_PROJECT_LIST),$(foreach locale,$(shell cat $(project)/locales/all-locales),$(foreach dir,$(LOCALES_$(project)),l10n/$(locale)/$(dir)))))
 else # MOZ_CO_LOCALES != all
 LOCALE_CO_DIRS = $(sort $(foreach locale,$(MOZ_CO_LOCALES),$(foreach dir,$(LOCALE_DIRS),l10n/$(locale)/$(dir))))
 endif
@@ -573,14 +573,14 @@
 ifdef RUN_AUTOCONF_LOCALLY
 	@echo "Removing local configures" ; \
 	cd $(ROOTDIR) && \
-	$(RM) -f mozilla/configure mozilla/nsprpub/configure \
-		mozilla/directory/c-sdk/configure
+	$(RM) -f configure nsprpub/configure \
+		directory/c-sdk/configure
 endif
 	@echo "checkout start: "`date` | tee $(CVSCO_LOGFILE)
-	@echo '$(CVSCO) $(CVS_CO_DATE_FLAGS) mozilla/client.mk $(MOZCONFIG_MODULES)'; \
+	@echo '$(CVSCO) $(CVS_CO_DATE_FLAGS) client.mk $(MOZCONFIG_MODULES)'; \
         cd $(ROOTDIR) && \
-	$(CVSCO) $(CVS_CO_DATE_FLAGS) mozilla/client.mk $(MOZCONFIG_MODULES)
-	@cd $(ROOTDIR) && $(MAKE) -f mozilla/client.mk real_checkout
+	$(CVSCO) $(CVS_CO_DATE_FLAGS) client.mk $(MOZCONFIG_MODULES)
+	@cd $(ROOTDIR) && $(MAKE) -f client.mk real_checkout
 
 #	Start the checkout. Split the output to the tty and a log file.
 
@@ -596,7 +596,7 @@
 	$(CHECKOUT_LOCALES);
 	@echo "checkout finish: "`date` | tee -a $(CVSCO_LOGFILE)
 # update the NSS checkout timestamp
-	@if test `egrep -c '^(U|C) mozilla/security/(nss|coreconf)' $(CVSCO_LOGFILE) 2>/dev/null` != 0; then \
+	@if test `egrep -c '^(U|C) security/(nss|coreconf)' $(CVSCO_LOGFILE) 2>/dev/null` != 0; then \
 		touch $(TOPSRCDIR)/security/manager/.nss.checkout; \
 	fi
 #	@: Check the log for conflicts. ;
@@ -624,13 +624,13 @@
 ifdef RUN_AUTOCONF_LOCALLY
 	@echo "Removing local configures" ; \
 	cd $(ROOTDIR) && \
-	$(RM) -f mozilla/configure mozilla/nsprpub/configure \
-		mozilla/directory/c-sdk/configure
+	$(RM) -f configure nsprpub/configure \
+		directory/c-sdk/configure
 endif
 	@echo "checkout start: "`date` | tee $(CVSCO_LOGFILE)
-	@echo '$(CVSCO) mozilla/client.mk $(MOZCONFIG_MODULES)'; \
+	@echo '$(CVSCO) client.mk $(MOZCONFIG_MODULES)'; \
         cd $(ROOTDIR) && \
-	$(CVSCO) mozilla/client.mk $(MOZCONFIG_MODULES)
+	$(CVSCO) client.mk $(MOZCONFIG_MODULES)
 	@cd $(TOPSRCDIR) && \
 	$(MAKE) -f client.mk real_fast-update
 
@@ -650,7 +650,7 @@
 	$(FASTUPDATE_LOCALES);
 	@echo "fast_update finish: "`date` | tee -a $(CVSCO_LOGFILE)
 # update the NSS checkout timestamp
-	@if test `egrep -c '^(U|C) mozilla/security/(nss|coreconf)' $(CVSCO_LOGFILE) 2>/dev/null` != 0; then \
+	@if test `egrep -c '^(U|C) security/(nss|coreconf)' $(CVSCO_LOGFILE) 2>/dev/null` != 0; then \
 		touch $(TOPSRCDIR)/security/manager/.nss.checkout; \
 	fi
 #	@: Check the log for conflicts. ;
