diff -burNp mozilla/toolkit/crashreporter/google-breakpad/src/common/linux/dump_symbols.cc mozilla-dud/toolkit/crashreporter/google-breakpad/src/common/linux/dump_symbols.cc
--- mozilla/toolkit/crashreporter/google-breakpad/src/common/linux/dump_symbols.cc	2008-02-25 19:18:09.000000000 +0100
+++ mozilla-dud/toolkit/crashreporter/google-breakpad/src/common/linux/dump_symbols.cc	2009-04-29 14:30:50.481579531 +0200
@@ -30,6 +30,7 @@
 #include <a.out.h>
 #include <cstdarg>
 #include <cstdlib>
+#include <cstdio>
 #include <cxxabi.h>
 #include <elf.h>
 #include <errno.h>
 
diff -burNp mozilla.orig/toolkit/crashreporter/google-breakpad/src/client/linux/handler/exception_handler.cc mozilla.dud/toolkit/crashreporter/google-breakpad/src/client/linux/handler/exception_handler.cc
--- mozilla.orig/toolkit/crashreporter/google-breakpad/src/client/linux/handler/exception_handler.cc	2008-02-25 19:18:08.000000000 +0100
+++ mozilla.dud/toolkit/crashreporter/google-breakpad/src/client/linux/handler/exception_handler.cc	2009-04-29 14:55:22.044889645 +0200
@@ -36,6 +36,7 @@
 
 #include <cassert>
 #include <cstdlib>
+#include <cstdio>
 #include <ctime>
 #include <linux/limits.h>
 
diff -burNp mozilla.orig/toolkit/crashreporter/google-breakpad/src/client/linux/handler/minidump_generator.cc mozilla.dud/toolkit/crashreporter/google-breakpad/src/client/linux/handler/minidump_generator.cc
--- mozilla.orig/toolkit/crashreporter/google-breakpad/src/client/linux/handler/minidump_generator.cc	2008-02-25 19:18:08.000000000 +0100
+++ mozilla.dud/toolkit/crashreporter/google-breakpad/src/client/linux/handler/minidump_generator.cc	2009-04-29 15:00:56.958685026 +0200
@@ -39,6 +39,7 @@
 #include <sys/wait.h>
 
 #include <cstdlib>
+#include <cstdio>
 #include <ctime>
 #include <string.h>
 
