diff -ruN mozilla-release.orig/browser/installer/package-manifest.in mozilla-release/browser/installer/package-manifest.in
--- mozilla-release.orig/browser/installer/package-manifest.in	2015-04-03 04:30:05.000000000 +0200
+++ mozilla-release/browser/installer/package-manifest.in	2015-04-12 16:46:15.409547636 +0200
@@ -44,14 +44,18 @@
 [@AB_CD@]
 @RESPATH@/browser/chrome/@AB_CD@@JAREXT@
 @RESPATH@/browser/chrome/@AB_CD@.manifest
+#ifndef LIBXUL_SDK
 @RESPATH@/chrome/@AB_CD@@JAREXT@
 @RESPATH@/chrome/@AB_CD@.manifest
+#endif
 @RESPATH@/browser/defaults/profile/bookmarks.html
 @RESPATH@/browser/defaults/profile/chrome/*
 @RESPATH@/browser/defaults/profile/localstore.rdf
 @RESPATH@/browser/defaults/profile/mimeTypes.rdf
+#ifndef LIBXUL_SDK
 @RESPATH@/dictionaries/*
 @RESPATH@/hyphenation/*
+#endif
 @RESPATH@/browser/@PREF_DIR@/firefox-l10n.js
 @RESPATH@/browser/searchplugins/*
 #ifdef XP_WIN32
@@ -63,7 +67,9 @@
 #endif
 
 [xpcom]
+#ifndef LIBXUL_SDK
 @RESPATH@/dependentlibs.list
+#endif
 #ifdef GKMEDIAS_SHARED_LIBRARY
 @BINPATH@/@DLL_PREFIX@gkmedias@DLL_SUFFIX@
 #endif
@@ -72,8 +78,10 @@
 @BINPATH@/@DLL_PREFIX@mozglue@DLL_SUFFIX@
 #endif
 #ifndef MOZ_STATIC_JS
+#ifndef LIBXUL_SDK
 @BINPATH@/@DLL_PREFIX@mozjs@DLL_SUFFIX@
 #endif
+#endif
 #ifdef MOZ_DMD
 @BINPATH@/@DLL_PREFIX@dmd@DLL_SUFFIX@
 #endif
@@ -84,17 +92,23 @@
 @BINPATH@/@DLL_PREFIX@plds4@DLL_SUFFIX@
 #endif
 #endif
+#ifndef LIBXUL_SDK
 #ifdef XP_MACOSX
 @BINPATH@/XUL
 #else
+#ifndef LIBXUL_SDK
 @BINPATH@/@DLL_PREFIX@xul@DLL_SUFFIX@
 #endif
+#endif
+#endif
 #ifdef XP_MACOSX
 @BINPATH@/@MOZ_CHILD_PROCESS_NAME@.app/
 @BINPATH@/@DLL_PREFIX@plugin_child_interpose@DLL_SUFFIX@
 #else
+#ifndef LIBXUL_SDK
 @BINPATH@/@MOZ_CHILD_PROCESS_NAME@
 #endif
+#endif
 #ifdef XP_WIN32
 @BINPATH@/plugin-hang-ui@BIN_SUFFIX@
 #if MOZ_PACKAGE_MSVC_DLLS
@@ -141,7 +155,9 @@
 #ifdef MOZ_UPDATER
 @RESPATH@/update-settings.ini
 #endif
+#ifndef LIBXUL_SDK
 @RESPATH@/platform.ini
+#endif
 #ifndef MOZ_NATIVE_SQLITE
 #ifndef MOZ_FOLD_LIBS
 @BINPATH@/@DLL_PREFIX@mozsqlite3@DLL_SUFFIX@
@@ -156,6 +172,7 @@
 
 ; [Components]
 @RESPATH@/browser/components/components.manifest
+#ifndef LIBXUL_SDK
 @RESPATH@/components/alerts.xpt
 #ifdef ACCESSIBILITY
 #ifdef XP_WIN32
@@ -167,10 +184,12 @@
 @RESPATH@/components/appstartup.xpt
 @RESPATH@/components/autocomplete.xpt
 @RESPATH@/components/autoconfig.xpt
+#endif
 @RESPATH@/components/browser-element.xpt
 @RESPATH@/browser/components/browsercompsbase.xpt
 @RESPATH@/browser/components/browser-feeds.xpt
 @RESPATH@/browser/components/browsermodules.manifest
+#ifndef LIBXUL_SDK
 @RESPATH@/components/caps.xpt
 @RESPATH@/components/chrome.xpt
 @RESPATH@/components/commandhandler.xpt
@@ -249,7 +268,9 @@
 @RESPATH@/components/filepicker.xpt
 #endif
 @RESPATH@/components/find.xpt
+#endif
 @RESPATH@/browser/components/fuel.xpt
+#ifndef LIBXUL_SDK
 @RESPATH@/components/gfx.xpt
 @RESPATH@/components/html5.xpt
 @RESPATH@/components/htmlparser.xpt
@@ -270,7 +291,9 @@
 @RESPATH@/components/layout_xul.xpt
 @RESPATH@/components/locale.xpt
 @RESPATH@/components/lwbrk.xpt
+#endif
 @RESPATH@/browser/components/migration.xpt
+#ifndef LIBXUL_SDK
 @RESPATH@/components/mimetype.xpt
 @RESPATH@/components/mozfind.xpt
 @RESPATH@/components/necko_about.xpt
@@ -307,12 +330,16 @@
 @RESPATH@/components/rdf.xpt
 @RESPATH@/components/satchel.xpt
 @RESPATH@/components/saxparser.xpt
+#endif
 @RESPATH@/browser/components/sessionstore.xpt
+#ifndef LIBXUL_SDK
 @RESPATH@/components/services-crypto-component.xpt
 #ifdef MOZ_CAPTIVEDETECT
 @RESPATH@/components/captivedetect.xpt
 #endif
+#endif
 @RESPATH@/browser/components/shellservice.xpt
+#ifndef LIBXUL_SDK
 @RESPATH@/components/shistory.xpt
 @RESPATH@/components/spellchecker.xpt
 @RESPATH@/components/storage.xpt
@@ -354,8 +381,10 @@
 @RESPATH@/components/xultmpl.xpt
 @RESPATH@/components/zipwriter.xpt
 @RESPATH@/components/telemetry.xpt
+#endif
 
 ; JavaScript components
+#ifndef LIBXUL_SDK
 @RESPATH@/components/ChromeNotifications.js
 @RESPATH@/components/ChromeNotifications.manifest
 @RESPATH@/components/ConsoleAPI.manifest
@@ -364,6 +393,7 @@
 @RESPATH@/components/BrowserElementParent.js
 @RESPATH@/components/FeedProcessor.manifest
 @RESPATH@/components/FeedProcessor.js
+#endif
 @RESPATH@/browser/components/BrowserFeeds.manifest
 @RESPATH@/browser/components/FeedConverter.js
 @RESPATH@/browser/components/FeedWriter.js
@@ -385,6 +415,7 @@
 @RESPATH@/browser/components/Experiments.manifest
 @RESPATH@/browser/components/ExperimentsService.js
 @RESPATH@/browser/components/translation.manifest
+#ifndef LIBXUL_SDK
 @RESPATH@/components/Downloads.manifest
 @RESPATH@/components/DownloadLegacy.js
 @RESPATH@/components/BrowserPageThumbs.manifest
@@ -436,12 +469,16 @@
 @RESPATH@/components/ProcessSingleton.manifest
 @RESPATH@/components/MainProcessSingleton.js
 @RESPATH@/components/ContentProcessSingleton.js
+#endif
 @RESPATH@/browser/components/nsSessionStore.manifest
 @RESPATH@/browser/components/nsSessionStartup.js
 @RESPATH@/browser/components/nsSessionStore.js
+#ifndef LIBXUL_SDK
 @RESPATH@/components/nsURLFormatter.manifest
 @RESPATH@/components/nsURLFormatter.js
+#endif
 @RESPATH@/browser/components/@DLL_PREFIX@browsercomps@DLL_SUFFIX@
+#ifndef LIBXUL_SDK
 @RESPATH@/components/txEXSLTRegExFunctions.manifest
 @RESPATH@/components/txEXSLTRegExFunctions.js
 @RESPATH@/components/toolkitplaces.manifest
@@ -476,6 +515,7 @@
 @RESPATH@/components/FormAutofillStartup.js
 @RESPATH@/components/contentAreaDropListener.manifest
 @RESPATH@/components/contentAreaDropListener.js
+#endif
 @RESPATH@/browser/components/BrowserProfileMigrators.manifest
 @RESPATH@/browser/components/ProfileMigrator.js
 @RESPATH@/browser/components/ChromeProfileMigrator.js
@@ -487,6 +527,7 @@
 #ifdef XP_MACOSX
 @RESPATH@/browser/components/SafariProfileMigrator.js
 #endif
+#ifndef LIBXUL_SDK
 #ifdef MOZ_ENABLE_DBUS
 @RESPATH@/components/@DLL_PREFIX@dbusservice@DLL_SUFFIX@
 #endif
@@ -612,19 +653,24 @@
 @RESPATH@/components/TestInterfaceJS.js
 @RESPATH@/components/TestInterfaceJS.manifest
 #endif
+#endif
 
 ; Modules
 @RESPATH@/browser/modules/*
+#ifndef LIBXUL_SDK
 @RESPATH@/modules/*
+#endif
 
 ; Safe Browsing
 #ifdef MOZ_URL_CLASSIFIER
+#ifndef LIBXUL_SDK
 @RESPATH@/components/nsURLClassifier.manifest
 @RESPATH@/components/nsUrlClassifierHashCompleter.js
 @RESPATH@/components/nsUrlClassifierListManager.js
 @RESPATH@/components/nsUrlClassifierLib.js
 @RESPATH@/components/url-classifier.xpt
 #endif
+#endif
 
 ; ANGLE GLES-on-D3D rendering library
 #ifdef MOZ_ANGLE_RENDERER
@@ -652,10 +698,12 @@
 #endif
 @RESPATH@/browser/extensions/{972ce4c6-7e08-4474-a285-3208198ce6fd}/install.rdf
 @RESPATH@/browser/extensions/{972ce4c6-7e08-4474-a285-3208198ce6fd}/icon.png
+#ifndef LIBXUL_SDK
 @RESPATH@/chrome/toolkit@JAREXT@
 @RESPATH@/chrome/toolkit.manifest
 @RESPATH@/chrome/recording.manifest
 @RESPATH@/chrome/recording/*
+#endif
 #ifdef MOZ_GTK
 @RESPATH@/browser/chrome/icons/default/default16.png
 @RESPATH@/browser/chrome/icons/default/default32.png
@@ -683,9 +731,11 @@
 ; All the pref files must be part of base to prevent migration bugs
 @RESPATH@/browser/@PREF_DIR@/firefox.js
 @RESPATH@/browser/@PREF_DIR@/firefox-branding.js
+#ifndef LIBXUL_SDK
 @RESPATH@/greprefs.js
 @RESPATH@/defaults/autoconfig/platform.js
 @RESPATH@/defaults/autoconfig/prefcalls.js
+#endif
 @RESPATH@/browser/defaults/profile/prefs.js
 
 #ifndef LIBXUL_SDK
@@ -695,7 +745,7 @@
 @RESPATH@/defaults/pref/channel-prefs.js
 #else
 ; For Fx-on-xr, channel-prefs lives with the app preferences. (Bug 762588)
-@RESPATH@/@PREF_DIR@/channel-prefs.js
+@RESPATH@/browser/@PREF_DIR@/channel-prefs.js
 #endif
 
 ; Services (gre) prefs
@@ -703,11 +753,14 @@
 @RESPATH@/defaults/pref/services-notifications.js
 #endif
 #ifdef MOZ_SERVICES_SYNC
+#ifndef LIBXUL_SDK
 @RESPATH@/defaults/pref/services-sync.js
 #endif
+#endif
 
 ; [Layout Engine Resources]
 ; Style Sheets, Graphics and other Resources used by the layout engine.
+#ifndef LIBXUL_SDK
 @RESPATH@/res/EditorOverride.css
 @RESPATH@/res/contenteditable.css
 @RESPATH@/res/designmode.css
@@ -745,11 +798,14 @@
 #ifdef XP_MACOSX
 @RESPATH@/res/MainMenu.nib/
 #endif
+#endif
 
 ; svg
+#ifndef LIBXUL_SDK
 @RESPATH@/res/svg.css
 @RESPATH@/components/dom_svg.xpt
 @RESPATH@/components/dom_smil.xpt
+#endif
 
 ; [Personal Security Manager]
 ;
@@ -770,11 +826,13 @@
 #endif
 @BINPATH@/@DLL_PREFIX@softokn3@DLL_SUFFIX@
 #endif
+#ifndef LIBXUL_SDK
 @RESPATH@/chrome/pippki@JAREXT@
 @RESPATH@/chrome/pippki.manifest
 @RESPATH@/components/pipboot.xpt
 @RESPATH@/components/pipnss.xpt
 @RESPATH@/components/pippki.xpt
+#endif
 
 ; For process sandboxing
 #if defined(MOZ_SANDBOX)
@@ -887,9 +945,11 @@
 @RESPATH@/metro/modules
 #endif
 
+#ifndef LIBXUL_SDK
 @RESPATH@/components/DataStore.manifest
 @RESPATH@/components/DataStoreImpl.js
 @RESPATH@/components/dom_datastore.xpt
+#endif
 
 ; Shutdown Terminator
 @RESPATH@/components/nsTerminatorTelemetry.js
