--- mozilla-firefox/browser/app/profile/firefox.js.orig	2007-03-21 17:21:55.000000000 +0100
+++ mozilla-firefox/browser/app/profile/firefox.js	2007-03-21 17:23:53.000000000 +0100
@@ -163,6 +163,9 @@
 
 pref("general.useragent.locale", "@AB_CD@");
 pref("general.skins.selectedSkin", "classic/1.0");
+pref("general.useragent.vendor", "PLD");
+pref("general.useragent.vendorSub", "3.0");
+pref("general.useragent.vendorComment", "Th");
 
 pref("general.smoothScroll", false);
 #ifdef UNIX_BUT_NOT_MAC
