diff -ur iceweasel-24.0.org/mozilla/browser/installer/package-manifest.in iceweasel-24.0/mozilla/browser/installer/package-manifest.in
--- iceweasel-24.0.org/mozilla/browser/installer/package-manifest.in	2013-09-11 01:14:52.000000000 +0200
+++ iceweasel-24.0/mozilla/browser/installer/package-manifest.in	2013-10-23 08:51:36.788646262 +0200
@@ -32,14 +32,18 @@
 [@AB_CD@]
 @BINPATH@/browser/chrome/@AB_CD@@JAREXT@
 @BINPATH@/browser/chrome/@AB_CD@.manifest
+#ifndef LIBXUL_SDK
 @BINPATH@/chrome/@AB_CD@@JAREXT@
 @BINPATH@/chrome/@AB_CD@.manifest
+#endif
 @BINPATH@/browser/defaults/profile/bookmarks.html
 @BINPATH@/browser/defaults/profile/chrome/*
 @BINPATH@/browser/defaults/profile/localstore.rdf
 @BINPATH@/browser/defaults/profile/mimeTypes.rdf
+#ifndef LIBXUL_SDK
 @BINPATH@/dictionaries/*
 @BINPATH@/hyphenation/*
+#endif
 @BINPATH@/browser/@PREF_DIR@/firefox-l10n.js
 @BINPATH@/browser/searchplugins/*
 #ifdef XP_WIN32
@@ -51,17 +55,23 @@
 #endif
 
 [xpcom]
+#ifndef LIBXUL_SDK
 @BINPATH@/dependentlibs.list
+#endif
 #ifdef XP_WIN32
 @BINPATH@/@DLL_PREFIX@gkmedias@DLL_SUFFIX@
 #endif
+#ifndef LIBXUL_SDK
 @BINPATH@/@DLL_PREFIX@mozalloc@DLL_SUFFIX@
+#endif
 #ifdef MOZ_SHARED_MOZGLUE
 @BINPATH@/@DLL_PREFIX@mozglue@DLL_SUFFIX@
 #endif
 #ifndef MOZ_STATIC_JS
+#ifndef LIBXUL_SDK
 @BINPATH@/@DLL_PREFIX@mozjs@DLL_SUFFIX@
 #endif
+#endif
 #ifndef MOZ_NATIVE_NSPR
 #ifndef MOZ_FOLD_LIBS
 @BINPATH@/@DLL_PREFIX@nspr4@DLL_SUFFIX@
@@ -72,14 +82,18 @@
 #ifdef XP_MACOSX
 @BINPATH@/XUL
 #else
+#ifndef LIBXUL_SDK
 @BINPATH@/@DLL_PREFIX@xul@DLL_SUFFIX@
 #endif
+#endif
 #ifdef XP_MACOSX
 @BINPATH@/@MOZ_CHILD_PROCESS_NAME@.app/
 @BINPATH@/@DLL_PREFIX@plugin_child_interpose@DLL_SUFFIX@
 #else
+#ifndef LIBXUL_SDK
 @BINPATH@/@MOZ_CHILD_PROCESS_NAME@
 #endif
+#endif
 #ifdef XP_WIN32
 @BINPATH@/plugin-hang-ui@BIN_SUFFIX@
 #ifndef MOZ_DEBUG
@@ -118,7 +132,9 @@
 #ifdef MOZ_UPDATER
 @BINPATH@/update-settings.ini
 #endif
+#ifndef LIBXUL_SDK
 @BINPATH@/platform.ini
+#endif
 #ifndef XP_OS2
 #ifndef MOZ_NATIVE_SQLITE
 #ifndef MOZ_FOLD_LIBS
@@ -132,25 +148,34 @@
 #ifdef XP_UNIX
 #ifndef XP_MACOSX
 @BINPATH@/run-mozilla.sh
+#ifndef LIBXUL_SDK
 @BINPATH@/mozilla-xremote-client
 #endif
 #endif
+#endif
 
 ; [Components]
 @BINPATH@/browser/components/components.manifest
+#ifndef LIBXUL_SDK
 @BINPATH@/components/alerts.xpt
+#endif
 #ifdef ACCESSIBILITY
 #ifdef XP_WIN32
 @BINPATH@/AccessibleMarshal.dll
 #endif
+#ifndef LIBXUL_SDK
 @BINPATH@/components/accessibility.xpt
 #endif
+#endif
+#ifndef LIBXUL_SDK
 @BINPATH@/components/appshell.xpt
 @BINPATH@/components/appstartup.xpt
 @BINPATH@/components/autocomplete.xpt
 @BINPATH@/components/autoconfig.xpt
+#endif
 @BINPATH@/browser/components/browsercompsbase.xpt
 @BINPATH@/browser/components/browser-feeds.xpt
+#ifndef LIBXUL_SDK
 @BINPATH@/components/caps.xpt
 @BINPATH@/components/chrome.xpt
 @BINPATH@/components/commandhandler.xpt
@@ -169,6 +194,7 @@
 @BINPATH@/components/dom_apps.xpt
 @BINPATH@/components/dom_base.xpt
 @BINPATH@/components/dom_system.xpt
+#endif
 #ifdef MOZ_B2G_RIL
 @BINPATH@/components/dom_telephony.xpt
 @BINPATH@/components/dom_voicemail.xpt
@@ -212,6 +212,7 @@
 #ifdef MOZ_B2G_BT
 @BINPATH@/components/dom_bluetooth.xpt
 #endif
+#ifndef LIBXUL_SDK
 @BINPATH@/components/dom_camera.xpt
 @BINPATH@/components/dom_canvas.xpt
 @BINPATH@/components/dom_contacts.xpt
@@ -216,6 +243,8 @@
 @BINPATH@/components/dom_xbl.xpt
 @BINPATH@/components/dom_xpath.xpt
 @BINPATH@/components/dom_xul.xpt
+#endif
+#ifndef LIBXUL_SDK
 #ifdef MOZ_GAMEPAD
 @BINPATH@/components/dom_gamepad.xpt
 #endif
@@ -231,7 +260,9 @@
 @BINPATH@/components/filepicker.xpt
 #endif
 @BINPATH@/components/find.xpt
+#endif
 @BINPATH@/browser/components/fuel.xpt
+#ifndef LIBXUL_SDK
 @BINPATH@/components/gfx.xpt
 @BINPATH@/components/html5.xpt
 @BINPATH@/components/htmlparser.xpt
@@ -254,7 +285,9 @@
 @BINPATH@/components/layout_xul.xpt
 @BINPATH@/components/locale.xpt
 @BINPATH@/components/lwbrk.xpt
+#endif
 @BINPATH@/browser/components/migration.xpt
+#ifndef LIBXUL_SDK
 @BINPATH@/components/mimetype.xpt
 @BINPATH@/components/mozfind.xpt
 @BINPATH@/components/necko_about.xpt
@@ -289,12 +322,16 @@
 @BINPATH@/components/rdf.xpt
 @BINPATH@/components/satchel.xpt
 @BINPATH@/components/saxparser.xpt
+#endif
 @BINPATH@/browser/components/sessionstore.xpt
+#ifndef LIBXUL_SDK
 @BINPATH@/components/services-crypto-component.xpt
+#endif
 #ifdef MOZ_CAPTIVEDETECT
 @BINPATH@/components/captivedetect.xpt
 #endif
 @BINPATH@/browser/components/shellservice.xpt
+#ifndef LIBXUL_SDK
 @BINPATH@/components/shistory.xpt
 @BINPATH@/components/spellchecker.xpt
 @BINPATH@/components/storage.xpt
@@ -312,9 +349,11 @@
 @BINPATH@/components/webBrowser_core.xpt
 @BINPATH@/components/webbrowserpersist.xpt
 @BINPATH@/components/widget.xpt
+#endif
 #ifdef XP_MACOSX
 @BINPATH@/components/widget_cocoa.xpt
 #endif
+#ifndef LIBXUL_SDK
 @BINPATH@/components/windowds.xpt
 @BINPATH@/components/windowwatcher.xpt
 @BINPATH@/components/xpcom_base.xpt
@@ -339,6 +378,7 @@
 @BINPATH@/components/BrowserElementParent.js
 @BINPATH@/components/FeedProcessor.manifest
 @BINPATH@/components/FeedProcessor.js
+#endif
 @BINPATH@/browser/components/BrowserFeeds.manifest
 @BINPATH@/browser/components/FeedConverter.js
 @BINPATH@/browser/components/FeedWriter.js
@@ -354,6 +394,7 @@
 @BINPATH@/browser/components/DownloadsStartup.js
 @BINPATH@/browser/components/DownloadsUI.js
 @BINPATH@/browser/components/BrowserPlaces.manifest
+#ifndef LIBXUL_SDK
 @BINPATH@/components/Downloads.manifest
 @BINPATH@/components/DownloadLegacy.js
 @BINPATH@/components/BrowserPageThumbs.manifest
@@ -381,31 +422,39 @@
 @BINPATH@/components/nsDownloadManagerUI.js
 @BINPATH@/components/NetworkGeolocationProvider.manifest
 @BINPATH@/components/NetworkGeolocationProvider.js
+#endif
 @BINPATH@/browser/components/nsSidebar.manifest
 @BINPATH@/browser/components/nsSidebar.js
+#ifndef LIBXUL_SDK
 @BINPATH@/components/extensions.manifest
 @BINPATH@/components/addonManager.js
 @BINPATH@/components/amContentHandler.js
 @BINPATH@/components/amWebInstallListener.js
 @BINPATH@/components/nsBlocklistService.js
+#endif
 #ifdef MOZ_UPDATER
 @BINPATH@/components/nsUpdateService.manifest
 @BINPATH@/components/nsUpdateService.js
 @BINPATH@/components/nsUpdateServiceStub.js
 #endif
+#ifndef LIBXUL_SDK
 @BINPATH@/components/nsUpdateTimerManager.manifest
 @BINPATH@/components/nsUpdateTimerManager.js
 @BINPATH@/components/pluginGlue.manifest
+#endif
 @BINPATH@/browser/components/nsSessionStore.manifest
 @BINPATH@/browser/components/nsSessionStartup.js
 @BINPATH@/browser/components/nsSessionStore.js
+#ifndef LIBXUL_SDK
 @BINPATH@/components/nsURLFormatter.manifest
 @BINPATH@/components/nsURLFormatter.js
+#endif
 #ifndef XP_OS2
 @BINPATH@/browser/components/@DLL_PREFIX@browsercomps@DLL_SUFFIX@
 #else
 @BINPATH@/browser/components/brwsrcmp@DLL_SUFFIX@
 #endif
+#ifndef LIBXUL_SDK
 @BINPATH@/components/txEXSLTRegExFunctions.manifest
 @BINPATH@/components/txEXSLTRegExFunctions.js
 @BINPATH@/components/toolkitplaces.manifest
@@ -414,7 +463,9 @@
 @BINPATH@/components/nsPlacesAutoComplete.manifest
 @BINPATH@/components/nsPlacesAutoComplete.js
 @BINPATH@/components/nsPlacesExpiration.js
+#endif
 @BINPATH@/browser/components/PlacesProtocolHandler.js
+#ifndef LIBXUL_SDK
 @BINPATH@/components/PlacesCategoriesStarter.js
 @BINPATH@/components/ColorAnalyzer.js
 @BINPATH@/components/PageThumbsProtocol.js
@@ -437,6 +488,7 @@
 @BINPATH@/components/contentSecurityPolicy.js
 @BINPATH@/components/contentAreaDropListener.manifest
 @BINPATH@/components/contentAreaDropListener.js
+#endif
 #ifdef MOZ_B2G_RIL
 @BINPATH@/components/RadioInterfaceLayer.manifest
 @BINPATH@/components/RadioInterfaceLayer.js
@@ -462,17 +514,24 @@
 @BINPATH@/browser/components/SafariProfileMigrator.js
 #endif
 #ifdef MOZ_ENABLE_DBUS
+#ifndef LIBXUL_SDK
 @BINPATH@/components/@DLL_PREFIX@dbusservice@DLL_SUFFIX@
 #endif
+#endif
 #ifdef MOZ_ENABLE_GNOME_COMPONENT
+#ifndef LIBXUL_SDK
 @BINPATH@/components/@DLL_PREFIX@mozgnome@DLL_SUFFIX@
 #endif
+#endif
 #ifdef MOZ_ENABLE_GNOMEVFS
 @BINPATH@/components/@DLL_PREFIX@nkgnomevfs@DLL_SUFFIX@
 #endif
 #if defined(MOZ_ENABLE_DBUS) || defined(MOZ_ENABLE_GNOME_COMPONENT) || defined(MOZ_ENABLE_GNOMEVFS)
+#ifndef LIBXUL_SDK
 @BINPATH@/components/components.manifest
 #endif
+#endif
+#ifndef LIBXUL_SDK
 @BINPATH@/components/nsINIProcessor.manifest
 @BINPATH@/components/nsINIProcessor.js
 @BINPATH@/components/nsPrompter.manifest
@@ -488,10 +547,12 @@
 @BINPATH@/components/SyncComponents.manifest
 @BINPATH@/components/Weave.js
 #endif
+#endif
 #ifdef MOZ_CAPTIVEDETECT
 @BINPATH@/components/CaptivePortalDetectComponents.manifest
 @BINPATH@/components/captivedetect.js
 #endif
+#ifndef LIBXUL_SDK
 @BINPATH@/components/servicesComponents.manifest
 @BINPATH@/components/cryptoComponents.manifest
 @BINPATH@/components/TelemetryPing.js
@@ -532,28 +593,37 @@
 @BINPATH@/components/PeerConnection.js
 @BINPATH@/components/PeerConnection.manifest
 #endif
+#endif
 
+#ifndef LIBXUL_SDK
 @BINPATH@/chrome/marionette@JAREXT@
 @BINPATH@/chrome/marionette.manifest
 @BINPATH@/components/MarionetteComponents.manifest
 @BINPATH@/components/marionettecomponent.js
+#endif
 
 #ifdef MOZ_WEBSPEECH
+#ifndef LIBXUL_SDK
 @BINPATH@/components/dom_webspeechsynth.xpt
 #endif
+#endif
 
 ; Modules
 @BINPATH@/browser/modules/*
+#ifndef LIBXUL_SDK
 @BINPATH@/modules/*
+#endif
 
 ; Safe Browsing
 #ifdef MOZ_URL_CLASSIFIER
+#ifndef LIBXUL_SDK
 @BINPATH@/components/nsURLClassifier.manifest
 @BINPATH@/components/nsUrlClassifierHashCompleter.js
 @BINPATH@/components/nsUrlClassifierListManager.js
 @BINPATH@/components/nsUrlClassifierLib.js
 @BINPATH@/components/url-classifier.xpt
 #endif
+#endif
 
 ; ANGLE GLES-on-D3D rendering library
 #ifdef MOZ_ANGLE_RENDERER
@@ -570,10 +640,12 @@
 @BINPATH@/browser/chrome/pdfjs/*
 @BINPATH@/browser/extensions/{972ce4c6-7e08-4474-a285-3208198ce6fd}/install.rdf
 @BINPATH@/browser/extensions/{972ce4c6-7e08-4474-a285-3208198ce6fd}/icon.png
+#ifndef LIBXUL_SDK
 @BINPATH@/chrome/toolkit@JAREXT@
 @BINPATH@/chrome/toolkit.manifest
 @BINPATH@/chrome/recording.manifest
 @BINPATH@/chrome/recording/*
+#endif
 #ifdef MOZ_GTK
 @BINPATH@/browser/chrome/icons/default/default16.png
 @BINPATH@/browser/chrome/icons/default/default32.png
@@ -596,9 +668,11 @@
 ; All the pref files must be part of base to prevent migration bugs
 @BINPATH@/browser/@PREF_DIR@/firefox.js
 @BINPATH@/browser/@PREF_DIR@/firefox-branding.js
+#ifndef LIBXUL_SDK
 @BINPATH@/greprefs.js
 @BINPATH@/defaults/autoconfig/platform.js
 @BINPATH@/defaults/autoconfig/prefcalls.js
+#endif
 @BINPATH@/browser/defaults/profile/prefs.js
 
 #ifndef LIBXUL_SDK
@@ -608,7 +682,7 @@
 @BINPATH@/defaults/pref/channel-prefs.js
 #else
 ; For Fx-on-xr, channel-prefs lives with the app preferences. (Bug 762588)
-@BINPATH@/@PREF_DIR@/channel-prefs.js
+@BINPATH@/browser/@PREF_DIR@/channel-prefs.js
 #endif
 
 ; Services (gre) prefs
@@ -616,11 +690,14 @@
 @BINPATH@/defaults/pref/services-notifications.js
 #endif
 #ifdef MOZ_SERVICES_SYNC
+#ifndef LIBXUL_SDK
 @BINPATH@/defaults/pref/services-sync.js
 #endif
+#endif
 
 ; [Layout Engine Resources]
 ; Style Sheets, Graphics and other Resources used by the layout engine.
+#ifndef LIBXUL_SDK
 @BINPATH@/res/EditorOverride.css
 @BINPATH@/res/contenteditable.css
 @BINPATH@/res/designmode.css
@@ -663,6 +740,7 @@
 @BINPATH@/res/svg.css
 @BINPATH@/components/dom_svg.xpt
 @BINPATH@/components/dom_smil.xpt
+#endif
 
 ; [Personal Security Manager]
 ;
@@ -683,11 +761,13 @@
 #endif
 @BINPATH@/@DLL_PREFIX@softokn3@DLL_SUFFIX@
 #endif
+#ifndef LIBXUL_SDK
 @BINPATH@/chrome/pippki@JAREXT@
 @BINPATH@/chrome/pippki.manifest
 @BINPATH@/components/pipboot.xpt
 @BINPATH@/components/pipnss.xpt
 @BINPATH@/components/pippki.xpt
+#endif
 
 ; for Solaris SPARC
 #ifdef SOLARIS
