diff -urN mozilla.orig/browser/app/application.ini mozilla/browser/app/application.ini
--- mozilla.orig/browser/app/application.ini	2008-07-29 13:24:18.000000000 +0200
+++ mozilla/browser/app/application.ini	2008-07-29 13:34:50.000000000 +0200
@@ -37,8 +37,8 @@
 
 #filter substitution
 [App]
-Vendor=Mozilla
-Name=Firefox
+Vendor=
+Name=Iceweasel
 Version=@APP_VERSION@
 BuildID=@GRE_BUILDID@
 Copyright=Copyright (c) 1998 - 2008 mozilla.org
diff -urN mozilla.orig/browser/app/firefox-branding.js mozilla/browser/app/firefox-branding.js
--- mozilla.orig/browser/app/firefox-branding.js	2008-07-29 13:24:18.000000000 +0200
+++ mozilla/browser/app/firefox-branding.js	2008-07-29 13:46:30.000000000 +0200
@@ -1,14 +1,15 @@
-pref("startup.homepage_override_url","http://www.mozilla.org/projects/%APP%/%VERSION%/whatsnew/");
-pref("startup.homepage_welcome_url","http://www.mozilla.org/projects/%APP%/%VERSION%/firstrun/");
+pref("browser.startup.homepage_override.mstone", "ignore");
+pref("startup.homepage_override_url","");
+pref("startup.homepage_welcome_url","");
 // URL user can browse to manually if for some reason all update installation
 // attempts fail.
-pref("app.update.url.manual", "http://www.mozilla.org/products/%APP%/");
+pref("app.update.url.manual", "http://www.mozilla.org/products/firefox/");
 // A default value for the "More information about this update" link
 // supplied in the "An update is available" page of the update wizard. 
-pref("app.update.url.details", "http://www.mozilla.org/projects/%APP%/");
+pref("app.update.url.details", "http://www.mozilla.org/projects/firefox/");
 
 // Release notes URL
-pref("app.releaseNotesURL", "http://www.mozilla.org/projects/%APP%/%VERSION%/releasenotes/");
+pref("app.releaseNotesURL", "");
 
 // Search codes belong only in builds with official branding
 pref("browser.search.param.yahoo-fr", "");
diff -urN mozilla.orig/browser/app/profile/firefox.js mozilla/browser/app/profile/firefox.js
--- mozilla.orig/browser/app/profile/firefox.js	2008-07-29 13:24:18.000000000 +0200
+++ mozilla/browser/app/profile/firefox.js	2008-07-29 13:46:50.000000000 +0200
@@ -70,12 +70,12 @@
 
 // Preferences for the Get Add-ons pane
 pref("extensions.getAddons.showPane", true);
-pref("extensions.getAddons.browseAddons", "https://%LOCALE%.add-ons.mozilla.com/%LOCALE%/%APP%");
+pref("extensions.getAddons.browseAddons", "https://%LOCALE%.add-ons.mozilla.com/%LOCALE%/firefox");
 pref("extensions.getAddons.maxResults", 5);
-pref("extensions.getAddons.recommended.browseURL", "https://%LOCALE%.add-ons.mozilla.com/%LOCALE%/%APP%/recommended");
-pref("extensions.getAddons.recommended.url", "https://services.addons.mozilla.org/%LOCALE%/%APP%/api/%API_VERSION%/list/featured/all/10/%OS%/%VERSION%");
-pref("extensions.getAddons.search.browseURL", "https://%LOCALE%.add-ons.mozilla.com/%LOCALE%/%APP%/search?q=%TERMS%");
-pref("extensions.getAddons.search.url", "https://services.addons.mozilla.org/%LOCALE%/%APP%/api/%API_VERSION%/search/%TERMS%/all/10/%OS%/%VERSION%");
+pref("extensions.getAddons.recommended.browseURL", "https://%LOCALE%.add-ons.mozilla.com/%LOCALE%/firefox/recommended");
+pref("extensions.getAddons.recommended.url", "https://services.addons.mozilla.org/%LOCALE%/firefox/api/%API_VERSION%/list/featured/all/10/%OS%/%VERSION%");
+pref("extensions.getAddons.search.browseURL", "https://%LOCALE%.add-ons.mozilla.com/%LOCALE%/firefox/search?q=%TERMS%");
+pref("extensions.getAddons.search.url", "https://services.addons.mozilla.org/%LOCALE%/firefox/api/%API_VERSION%/search/%TERMS%/all/10/%OS%/%VERSION%");
 
 // Blocklist preferences
 pref("extensions.blocklist.enabled", true);
@@ -158,9 +158,9 @@
 pref("extensions.update.interval", 86400);  // Check for updates to Extensions and 
                                             // Themes every day
 // Non-symmetric (not shared by extensions) extension-specific [update] preferences
-pref("extensions.getMoreExtensionsURL", "https://%LOCALE%.add-ons.mozilla.com/%LOCALE%/%APP%/%VERSION%/extensions/");
-pref("extensions.getMoreThemesURL", "https://%LOCALE%.add-ons.mozilla.com/%LOCALE%/%APP%/%VERSION%/themes/");
-pref("extensions.getMorePluginsURL", "https://%LOCALE%.add-ons.mozilla.com/%LOCALE%/%APP%/%VERSION%/plugins/");
+pref("extensions.getMoreExtensionsURL", "https://%LOCALE%.add-ons.mozilla.com/%LOCALE%/firefox/%VERSION%/extensions/");
+pref("extensions.getMoreThemesURL", "https://%LOCALE%.add-ons.mozilla.com/%LOCALE%/firefox/%VERSION%/themes/");
+pref("extensions.getMorePluginsURL", "https://%LOCALE%.add-ons.mozilla.com/%LOCALE%/firefox/%VERSION%/plugins/");
 pref("extensions.dss.enabled", false);          // Dynamic Skin Switching                                               
 pref("extensions.dss.switchPending", false);    // Non-dynamic switch pending after next
                                                 // restart.
@@ -659,9 +659,7 @@
 // if we rev the EULA again, we should bump this so users agree to the new EULA
 pref("browser.EULA.version", 3);
 
-#ifdef DEBUG
 pref("browser.EULA.override", true);
-#endif
 
 pref("browser.sessionstore.enabled", true);
 pref("browser.sessionstore.resume_from_crash", true);
@@ -762,4 +760,4 @@
 pref("breakpad.reportURL", "http://crash-stats.mozilla.com/report/index/");
 
 // base URL for web-based support pages
-pref("app.support.baseURL", "http://support.mozilla.com/1/%APP%/%VERSION%/%OS%/%LOCALE%/");
+pref("app.support.baseURL", "http://support.mozilla.com/1/firefox/%VERSION%/%OS%/%LOCALE%/");
diff -urN mozilla.orig/browser/base/content/baseMenuOverlay.xul mozilla/browser/base/content/baseMenuOverlay.xul
--- mozilla.orig/browser/base/content/baseMenuOverlay.xul	2008-07-29 13:24:17.000000000 +0200
+++ mozilla/browser/base/content/baseMenuOverlay.xul	2008-07-29 13:34:50.000000000 +0200
@@ -92,17 +92,6 @@
                   accesskey="&helpForIEUsers.accesskey;"
                   oncommand="openHelpLink('ieusers');"/>
 #endif
-        <menuitem id="releaseNotes"
-                  accesskey="&helpReleaseNotes.accesskey;"
-                  label="&helpReleaseNotes.label;"
-                  oncommand="openReleaseNotes(event)"
-                  onclick="checkForMiddleClick(this, event);"/>
-        <menuseparator id="updateSeparator"/>
-        <menuitem id="checkForUpdates"
-                  accesskey="&updateCmd.accesskey;"
-                  label="&updateCmd.label;"
-                  class="menuitem-iconic"
-                  oncommand="checkForUpdates();"/>
         <menuseparator id="aboutSeparator"/>
         <menuitem id="aboutName"
                   accesskey="&aboutCmd.accesskey;"
diff -urN mozilla.orig/browser/base/content/credits.xhtml mozilla/browser/base/content/credits.xhtml
--- mozilla.orig/browser/base/content/credits.xhtml	2008-07-29 13:24:17.000000000 +0200
+++ mozilla/browser/base/content/credits.xhtml	2008-07-29 13:34:50.000000000 +0200
@@ -206,7 +206,7 @@
     </div>
 
     <div id="creditsBox">
-        <h2 class="title">&brandFullName;&trade;
+        <h2 class="title">&brandFullName;
           <div class="motto">&brandMotto;</div>
         </h2>
         <div class="links">
@@ -529,20 +529,13 @@
         <p id="gecko" class="center">&credit.poweredByGeckoReg;</p>
 
         <p class="footnote">
-          &brandFullName;&trade; &license.part0; &copy;1998-2008 &license.part1;
+          &brandFullName; &license.part0; &copy;1998-2008 &license.part1;
           <a href="" link="about:credits" onclick="visitLink(event);">&license.contrib;</a>,
           &license.part2;
           <a href="" link="about:license" onclick="visitLink(event);">about:license</a>
           &license.part3;</p>
 
         <p class="footnote">
-          Mozilla Firefox&reg; and the Firefox logo are registered trademarks of the
-          Mozilla Foundation. You are not granted rights or licenses to the trademarks
-          of the Mozilla Foundation or any party, including without limitation the
-          Firefox name or logo. Gecko&reg; is a registered trademark of Netscape
-          Communications Corporation.</p>
-
-        <p class="footnote">
           U.S. GOVERNMENT END USERS. The Software is a &ldquo;commercial item,&rdquo;
           as that term is defined in 48 C.F.R. 2.101 (Oct. 1995), consisting of
           &ldquo;commercial computer software&rdquo; and &ldquo;commercial computer software
diff -urN mozilla.orig/browser/branding/unofficial/pref/firefox-branding.js mozilla/browser/branding/unofficial/pref/firefox-branding.js
--- mozilla.orig/browser/branding/unofficial/pref/firefox-branding.js	2008-07-29 13:24:19.000000000 +0200
+++ mozilla/browser/branding/unofficial/pref/firefox-branding.js	2008-07-29 13:46:22.000000000 +0200
@@ -1,14 +1,14 @@
-pref("startup.homepage_override_url","http://www.mozilla.org/projects/%APP%/%VERSION%/whatsnew/");
-pref("startup.homepage_welcome_url","http://www.mozilla.org/projects/%APP%/%VERSION%/firstrun/");
+pref("startup.homepage_override_url","http://www.mozilla.org/projects/firefox/%VERSION%/whatsnew/");
+pref("startup.homepage_welcome_url","http://www.mozilla.org/projects/firefox/%VERSION%/firstrun/");
 // URL user can browse to manually if for some reason all update installation
 // attempts fail.
-pref("app.update.url.manual", "http://www.mozilla.org/products/%APP%/");
+pref("app.update.url.manual", "http://www.mozilla.org/products/firefox/");
 // A default value for the "More information about this update" link
 // supplied in the "An update is available" page of the update wizard. 
-pref("app.update.url.details", "http://www.mozilla.org/projects/%APP%/");
+pref("app.update.url.details", "http://www.mozilla.org/projects/firefox/");
 
 // Release notes URL
-pref("app.releaseNotesURL", "http://www.mozilla.org/projects/%APP%/%VERSION%/releasenotes/");
+pref("app.releaseNotesURL", "http://www.mozilla.org/projects/firefox/%VERSION%/releasenotes/");
 
 // Search codes belong only in builds with official branding
 pref("browser.search.param.yahoo-fr", "");
diff -urN mozilla.orig/browser/components/nsBrowserContentHandler.js mozilla/browser/components/nsBrowserContentHandler.js
--- mozilla.orig/browser/components/nsBrowserContentHandler.js	2008-07-29 13:24:18.000000000 +0200
+++ mozilla/browser/components/nsBrowserContentHandler.js	2008-07-29 13:34:50.000000000 +0200
@@ -836,7 +836,7 @@
   },
 
   // XXX localize me... how?
-  helpInfo : "Usage: firefox [-flags] [<url>]\n",
+  helpInfo : "Usage: iceweasel [-flags] [<url>]\n",
 
   /* nsIFactory */
   createInstance: function dch_CI(outer, iid) {
diff -urN mozilla.orig/browser/components/nsBrowserGlue.js mozilla/browser/components/nsBrowserGlue.js
--- mozilla.orig/browser/components/nsBrowserGlue.js	2008-07-29 13:24:18.000000000 +0200
+++ mozilla/browser/components/nsBrowserGlue.js	2008-07-29 13:34:50.000000000 +0200
@@ -759,7 +759,7 @@
   },
 
   // for XPCOM
-  classDescription: "Firefox Browser Glue Service",
+  classDescription: "Iceweasel Browser Glue Service",
   classID:          Components.ID("{eab9012e-5f74-4cbc-b2b5-a590235513cc}"),
   contractID:       "@mozilla.org/browser/browserglue;1",
 
diff -urN mozilla.orig/browser/components/preferences/advanced.xul mozilla/browser/components/preferences/advanced.xul
--- mozilla.orig/browser/components/preferences/advanced.xul	2008-07-29 13:24:18.000000000 +0200
+++ mozilla/browser/components/preferences/advanced.xul	2008-07-29 13:34:50.000000000 +0200
@@ -267,10 +267,6 @@
         <tabpanel id="updatePanel" orient="vertical" align="start">
           <label control="autoUpdateGroup">&autoCheck.label;</label>
           <vbox class="indent" id="autoUpdateGroup" role="group">
-            <checkbox id="enableAppUpdate"
-                      label="&enableAppUpdate.label;"
-                      accesskey="&enableAppUpdate.accesskey;"
-                      preference="app.update.enabled"/>
             <checkbox id="enableAddonUpdate"
                       label="&enableAddonsUpdate.label;"
                       accesskey="&enableAddonsUpdate.accesskey;"
diff -urN mozilla.orig/browser/components/preferences/applications.js mozilla/browser/components/preferences/applications.js
--- mozilla.orig/browser/components/preferences/applications.js	2008-07-29 13:24:18.000000000 +0200
+++ mozilla/browser/components/preferences/applications.js	2008-07-29 13:34:50.000000000 +0200
@@ -538,8 +538,7 @@
 
   _shellSvc:
 #ifdef HAVE_SHELL_SERVICE
-    Cc["@mozilla.org/browser/shell-service;1"].
-    getService(Ci.nsIShellService),
+    getShellService(),
 #else
     null,
 #endif
diff -urN mozilla.orig/browser/components/safebrowsing/content/report-phishing-overlay.xul mozilla/browser/components/safebrowsing/content/report-phishing-overlay.xul
--- mozilla.orig/browser/components/safebrowsing/content/report-phishing-overlay.xul	2008-07-29 13:24:18.000000000 +0200
+++ mozilla/browser/components/safebrowsing/content/report-phishing-overlay.xul	2008-07-29 13:34:50.000000000 +0200
@@ -50,7 +50,7 @@
     <menuitem id="menu_HelpPopup_reportPhishingtoolmenu"
               label="&reportPhishSiteMenu.title2;"
               accesskey="&reportPhishSiteMenu.accesskey;"
-              insertbefore="updateSeparator"
+              insertbefore="aboutSeparator"
               observes="reportPhishingBroadcaster"
               oncommand="openUILink(safebrowsing.getReportURL('Phish'), event);"
               onclick="checkForMiddleClick(this, event);"/>
diff -urN mozilla.orig/browser/components/shell/src/nsWindowsShellService.cpp mozilla/browser/components/shell/src/nsWindowsShellService.cpp
--- mozilla.orig/browser/components/shell/src/nsWindowsShellService.cpp	2008-07-29 13:24:18.000000000 +0200
+++ mozilla/browser/components/shell/src/nsWindowsShellService.cpp	2008-07-29 13:34:50.000000000 +0200
@@ -189,7 +189,7 @@
   PRInt32 flags;
 } SETTING;
 
-#define APP_REG_NAME L"Firefox"
+#define APP_REG_NAME L"Iceweasel"
 #define DI "\\DefaultIcon"
 #define SOP "\\shell\\open\\command"
 
diff -urN mozilla.orig/browser/confvars.sh mozilla/browser/confvars.sh
--- mozilla.orig/browser/confvars.sh	2008-07-29 13:24:18.000000000 +0200
+++ mozilla/browser/confvars.sh	2008-07-29 13:34:50.000000000 +0200
@@ -36,7 +36,7 @@
 #
 # ***** END LICENSE BLOCK *****
 
-MOZ_APP_NAME=firefox
+MOZ_APP_NAME=iceweasel
 MOZ_UPDATER=1
 MOZ_PHOENIX=1
 
diff -urN mozilla.orig/browser/locales/en-US/chrome/browser/aboutDialog.dtd mozilla/browser/locales/en-US/chrome/browser/aboutDialog.dtd
--- mozilla.orig/browser/locales/en-US/chrome/browser/aboutDialog.dtd	2008-07-29 13:24:18.000000000 +0200
+++ mozilla/browser/locales/en-US/chrome/browser/aboutDialog.dtd	2008-07-29 13:34:50.000000000 +0200
@@ -4,6 +4,4 @@
 <!ENTITY aboutLink              "&lt; About &brandFullName;">
 <!ENTITY aboutLink.accesskey    "A">
 <!ENTITY aboutVersion           "version">
-<!ENTITY copyrightInfo          "&#169;1998-2008 Contributors. All Rights Reserved. Firefox and the 
-                                 Firefox logos are trademarks of the Mozilla Foundation.  All rights 
-                                 reserved.">
+<!ENTITY copyrightInfo          "&#169;1998-2008 Contributors. All Rights Reserved.">
diff -urN mozilla.orig/browser/locales/en-US/chrome/overrides/appstrings.properties mozilla/browser/locales/en-US/chrome/overrides/appstrings.properties
--- mozilla.orig/browser/locales/en-US/chrome/overrides/appstrings.properties	2008-07-29 13:24:18.000000000 +0200
+++ mozilla/browser/locales/en-US/chrome/overrides/appstrings.properties	2008-07-29 13:34:50.000000000 +0200
@@ -35,23 +35,23 @@
 # ***** END LICENSE BLOCK *****
 
 malformedURI=The URL is not valid and cannot be loaded.
-fileNotFound=Firefox can't find the file at %S.
-dnsNotFound=Firefox can't find the server at %S.
-protocolNotFound=Firefox doesn't know how to open this address, because the protocol (%S) isn't associated with any program.
-connectionFailure=Firefox can't establish a connection to the server at %S.
+fileNotFound=Iceweasel can't find the file at %S.
+dnsNotFound=Iceweasel can't find the server at %S.
+protocolNotFound=Iceweasel doesn't know how to open this address, because the protocol (%S) isn't associated with any program.
+connectionFailure=Iceweasel can't establish a connection to the server at %S.
 netInterrupt=The connection to %S was interrupted while the page was loading.
 netTimeout=The server at %S is taking too long to respond.
-redirectLoop=Firefox has detected that the server is redirecting the request for this address in a way that will never complete.
+redirectLoop=Iceweasel has detected that the server is redirecting the request for this address in a way that will never complete.
 ## LOCALIZATION NOTE (confirmRepostPrompt): In this item, don't translate "%S"
 confirmRepostPrompt=To display this page, %S must send information that will repeat any action (such as a search or order confirmation) that was performed earlier.
 resendButton.label=Resend
-unknownSocketType=Firefox doesn't know how to communicate with the server.
+unknownSocketType=Iceweasel doesn't know how to communicate with the server.
 netReset=The connection to the server was reset while the page was loading.
-netOffline=Firefox is currently in offline mode and can't browse the Web.
+netOffline=Iceweasel is currently in offline mode and can't browse the Web.
 isprinting=The document cannot change while Printing or in Print Preview.
-deniedPortAccess=This address uses a network port which is normally used for purposes other than Web browsing. Firefox has canceled the request for your protection.
-proxyResolveFailure=Firefox is configured to use a proxy server that can't be found.
-proxyConnectFailure=Firefox is configured to use a proxy server that is refusing connections.
+deniedPortAccess=This address uses a network port which is normally used for purposes other than Web browsing. Iceweasel has canceled the request for your protection.
+proxyResolveFailure=Iceweasel is configured to use a proxy server that can't be found.
+proxyConnectFailure=Iceweasel is configured to use a proxy server that is refusing connections.
 contentEncodingError=The page you are trying to view cannot be shown because it uses an invalid or unsupported form of compression.
 unsafeContentType=The page you are trying to view cannot be shown because it is contained in a file type that may not be safe to open. Please contact the website owners to inform them of this problem.
 externalProtocolTitle=External Protocol Request
diff -urN mozilla.orig/browser/locales/en-US/searchplugins/answers.xml mozilla/browser/locales/en-US/searchplugins/answers.xml
--- mozilla.orig/browser/locales/en-US/searchplugins/answers.xml	2008-07-29 13:24:18.000000000 +0200
+++ mozilla/browser/locales/en-US/searchplugins/answers.xml	2008-07-29 13:34:50.000000000 +0200
@@ -8,6 +8,6 @@
   <Param name="gwp" value="13"/>
 </Url>
 <Url type="application/x-suggestions+json" method="GET"
-     template="http://www.answers.com/main/startswith?output=json&amp;client=firefox&amp;s={searchTerms}"/>
+     template="http://www.answers.com/main/startswith?output=json&amp;client=iceweasel&amp;s={searchTerms}"/>
 <SearchForm>http://www.answers.com/</SearchForm>
 </SearchPlugin>
diff -urN mozilla.orig/browser/locales/en-US/searchplugins/google.xml mozilla/browser/locales/en-US/searchplugins/google.xml
--- mozilla.orig/browser/locales/en-US/searchplugins/google.xml	2008-07-29 13:24:18.000000000 +0200
+++ mozilla/browser/locales/en-US/searchplugins/google.xml	2008-07-29 13:34:51.000000000 +0200
@@ -3,7 +3,7 @@
 <Description>Google Search</Description>
 <InputEncoding>UTF-8</InputEncoding>
 <Image width="16" height="16">data:image/x-icon;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAIAAACQkWg2AAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAAaRJREFUeNpiVIg5JRURw0A0YAHio943kYV%2B%2Ff33%2BdvvX7%2F%2FMjEx8nKycrGzwKXOiPKzICvdeezLhCV3jp15%2Bfv%2FX0YGhv8MDDxMX2qKTIw0RK10eYD6QYqATvoPBkt3f5K0W9Ew4fjTFz%2F%2Bw8Dm3W8UPeZxqFa%2BevsFyD0twgfVsOfkRxHrtfV9u5BVQ8Crd98%2FffkGYQM1QJ20%2FfSPv79eNxQGYfpSVJADmcvEAHbr7oOX2dj%2FERNKIA2%2F%2F%2Fz%2FxfCDhYVoDUDw5P6vf9%2B5iY0HVmZGQWm%2BN3fff%2Fn2k4eLHS739x%2FDiRs%2Ff%2F%2F5x8HO%2FOHzN3djfqgNjIwMgc6qzLx%2Fpy47j2zY%2Feff06tXhOUucgxeun33AUZGpHh4%2Bvo7t8EyIJqz%2FhpasD59%2B5dNrqdnznZIsEL9ICXCsWuBCwvTv%2FymS5PWPP32ExEALz%2F%2BB5r848cPCJcRaMP9xaYQzofPPzfuvrnj0Jst%2B5%2F8%2Bc4sLPeDkYlRgJc93VPE18NIXkYUmJYQSQMZ%2FP3379uPH7%2F%2F%2FEETBzqJ0WqLGvFpe2LCC4AAAwAyjg7ENzDDWAAAAABJRU5ErkJggg%3D%3D</Image>
-<Url type="application/x-suggestions+json" method="GET" template="http://suggestqueries.google.com/complete/search?output=firefox&amp;client=firefox&amp;hl={moz:locale}&amp;q={searchTerms}"/>
+<Url type="application/x-suggestions+json" method="GET" template="http://suggestqueries.google.com/complete/search?output=firefox&amp;client=iceweasel&amp;hl={moz:locale}&amp;q={searchTerms}"/>
 <Url type="text/html" method="GET" template="http://www.google.com/search">
   <Param name="q" value="{searchTerms}"/>
   <Param name="ie" value="utf-8"/>
@@ -11,7 +11,7 @@
   <Param name="aq" value="t"/>
   <!-- Dynamic parameters -->
   <Param name="rls" value="{moz:distributionID}:{moz:locale}:{moz:official}"/>
-  <MozParam name="client" condition="defaultEngine" trueValue="firefox-a" falseValue="firefox"/>
+  <MozParam name="client" condition="defaultEngine" trueValue="iceweasel-a" falseValue="iceweasel"/>
 </Url>
-<SearchForm>http://www.google.com/firefox</SearchForm>
+<SearchForm>http://www.google.com/</SearchForm>
 </SearchPlugin>
diff -urN mozilla.orig/browser/locales/en-US/updater/updater.ini mozilla/browser/locales/en-US/updater/updater.ini
--- mozilla.orig/browser/locales/en-US/updater/updater.ini	2008-07-29 13:24:18.000000000 +0200
+++ mozilla/browser/locales/en-US/updater/updater.ini	2008-07-29 13:34:51.000000000 +0200
@@ -1,4 +1,4 @@
 ; This file is in the UTF-8 encoding
 [Strings]
 Title=Software Update
-Info=Firefox is installing your updates and will start in a few moments…
+Info=Iceweasel is installing your updates and will start in a few moments…
diff -urN mozilla.orig/extensions/reporter/resources/content/reporter/reporterOverlay.xul mozilla/extensions/reporter/resources/content/reporter/reporterOverlay.xul
--- mozilla.orig/extensions/reporter/resources/content/reporter/reporterOverlay.xul	2008-07-29 13:24:30.000000000 +0200
+++ mozilla/extensions/reporter/resources/content/reporter/reporterOverlay.xul	2008-07-29 13:34:51.000000000 +0200
@@ -53,7 +53,7 @@
     <menuitem id="menu_HelpPopup_reportertoolmenu" 
               label="&reporterMenu2.title;" 
               accesskey="&reporterMenu.accesskey;"
-              insertafter="releaseNotes"
+              insertbefore="aboutSeparator"
               observes="reporterItemsBroadcaster"
               oncommand="loadReporterWizard()"/>
   </menupopup>
diff -urN mozilla.orig/toolkit/xre/nsAppRunner.cpp mozilla/toolkit/xre/nsAppRunner.cpp
--- mozilla.orig/toolkit/xre/nsAppRunner.cpp	2008-07-29 13:25:13.000000000 +0200
+++ mozilla/toolkit/xre/nsAppRunner.cpp	2008-07-29 13:34:51.000000000 +0200
@@ -1184,8 +1184,7 @@
 static inline void
 DumpVersion()
 {
-  printf("%s %s %s", 
-         gAppData->vendor ? gAppData->vendor : "", gAppData->name, gAppData->version);
+  printf("%s %s", gAppData->name, gAppData->version);
   if (gAppData->copyright)
       printf(", %s", gAppData->copyright);
   printf("\n");
