--- iceweasel-3.0.10/mozilla/build/unix/run-mozilla.sh~	2007-09-04 20:09:31.000000000 +0300
+++ iceweasel-3.0.10/mozilla/build/unix/run-mozilla.sh	2009-05-05 19:20:55.718561642 +0300
@@ -438,7 +438,7 @@
 then
 	moz_debug_program ${1+"$@"}
 else
-	moz_run_program ${1+"$@"}
+	exec "$MOZ_PROGRAM" ${1+"$@"}
 fi
 
 exit $exitcode
--- iceweasel-3.0.10/mozilla/build/unix/mozilla.in~	2007-10-05 10:29:26.000000000 +0300
+++ iceweasel-3.0.10/mozilla/build/unix/mozilla.in	2009-05-05 19:26:08.278322082 +0300
@@ -135,8 +135,5 @@
 then
   echo $dist_bin/run-mozilla.sh $script_args $dist_bin/$MOZILLA_BIN "$@"
 fi
-"$dist_bin/run-mozilla.sh" $script_args "$dist_bin/$MOZILLA_BIN" "$@"
-exitcode=$?
-
-exit $exitcode
+exec "$dist_bin/run-mozilla.sh" $script_args "$dist_bin/$MOZILLA_BIN" "$@"
 # EOF.
