--- mozilla.orig/modules/libpref/src/init/all.js	2006-05-08 22:22:51.000000000 +0200
+++ mozilla/modules/libpref/src/init/all.js	2006-07-25 07:44:06.744790000 +0200
@@ -1886,33 +1886,33 @@
 
 // th
 
-pref("font.name.serif.tr", "Times");
-pref("font.name.sans-serif.tr", "Helvetica");
-pref("font.name.monospace.tr", "Courier");
+pref("font.name.serif.tr", "serif");
+pref("font.name.sans-serif.tr", "sans-serif");
+pref("font.name.monospace.tr", "monospace");
 
 pref("font.name.serif.x-baltic", "serif");
 pref("font.name.sans-serif.x-baltic", "sans-serif");
 pref("font.name.monospace.x-baltic", "monospace");
 
-pref("font.name.serif.x-central-euro", "Times");
-pref("font.name.sans-serif.x-central-euro", "Helvetica");
-pref("font.name.monospace.x-central-euro", "Courier");
+pref("font.name.serif.x-central-euro", "serif");
+pref("font.name.sans-serif.x-central-euro", "sans-serif");
+pref("font.name.monospace.x-central-euro", "monospace");
 
 pref("font.name.serif.x-cyrillic", "serif");
 pref("font.name.sans-serif.x-cyrillic", "sans-serif");
 pref("font.name.monospace.x-cyrillic", "monospace");
 
-pref("font.name.serif.x-unicode", "Times");
-pref("font.name.sans-serif.x-unicode", "Helvetica");
-pref("font.name.monospace.x-unicode", "Courier");
-
-pref("font.name.serif.x-user-def", "Times");
-pref("font.name.sans-serif.x-user-def", "Helvetica");
-pref("font.name.monospace.x-user-def", "Courier");
-
-pref("font.name.serif.x-western", "Times");
-pref("font.name.sans-serif.x-western", "Helvetica");
-pref("font.name.monospace.x-western", "Courier");
+pref("font.name.serif.x-unicode", "serif");
+pref("font.name.sans-serif.x-unicode", "sans-serif");
+pref("font.name.monospace.x-unicode", "monospace");
+
+pref("font.name.serif.x-user-def", "serif");
+pref("font.name.sans-serif.x-user-def", "sans-serif");
+pref("font.name.monospace.x-user-def", "monospace");
+
+pref("font.name.serif.x-western", "serif");
+pref("font.name.sans-serif.x-western", "sans-serif");
+pref("font.name.monospace.x-western", "monospace");
 
 pref("font.name.serif.zh-CN", "serif");
 pref("font.name.sans-serif.zh-CN", "sans-serif");
@@ -1989,63 +1989,63 @@
 
 pref("font.default.ar", "sans-serif");
 pref("font.size.variable.ar", 16);
-pref("font.size.fixed.ar", 12);
+pref("font.size.fixed.ar", 14);
 
-pref("font.default.el", "serif");
+pref("font.default.el", "sans-serif");
 pref("font.size.variable.el", 16);
-pref("font.size.fixed.el", 12);
+pref("font.size.fixed.el", 14);
 
 pref("font.default.he", "sans-serif");
 pref("font.size.variable.he", 16);
-pref("font.size.fixed.he", 12);
+pref("font.size.fixed.he", 14);
 
 pref("font.default.ja", "sans-serif");
 pref("font.size.variable.ja", 16);
-pref("font.size.fixed.ja", 16);
+pref("font.size.fixed.ja", 14);
 
 pref("font.default.ko", "sans-serif");
 pref("font.size.variable.ko", 16);
-pref("font.size.fixed.ko", 16);
+pref("font.size.fixed.ko", 14);
 
-pref("font.default.th", "serif");
+pref("font.default.th", "sans-serif");
 pref("font.size.variable.th", 16);
-pref("font.size.fixed.th", 12);
+pref("font.size.fixed.th", 14);
 
-pref("font.default.tr", "serif");
+pref("font.default.tr", "sans-serif");
 pref("font.size.variable.tr", 16);
-pref("font.size.fixed.tr", 12);
+pref("font.size.fixed.tr", 14);
 
-pref("font.default.x-baltic", "serif");
+pref("font.default.x-baltic", "sans-serif");
 pref("font.size.variable.x-baltic", 16);
-pref("font.size.fixed.x-baltic", 12);
+pref("font.size.fixed.x-baltic", 14);
 
-pref("font.default.x-central-euro", "serif");
+pref("font.default.x-central-euro", "sans-serif");
 pref("font.size.variable.x-central-euro", 16);
-pref("font.size.fixed.x-central-euro", 12);
+pref("font.size.fixed.x-central-euro", 14);
 
-pref("font.default.x-cyrillic", "serif");
+pref("font.default.x-cyrillic", "sans-serif");
 pref("font.size.variable.x-cyrillic", 16);
-pref("font.size.fixed.x-cyrillic", 12);
+pref("font.size.fixed.x-cyrillic", 14);
 
-pref("font.default.x-unicode", "serif");
+pref("font.default.x-unicode", "sans-serif");
 pref("font.size.variable.x-unicode", 16);
-pref("font.size.fixed.x-unicode", 12);
+pref("font.size.fixed.x-unicode", 14);
 
-pref("font.default.x-western", "serif");
+pref("font.default.x-western", "sans-serif");
 pref("font.size.variable.x-western", 16);
-pref("font.size.fixed.x-western", 12);
+pref("font.size.fixed.x-western", 14);
 
 pref("font.default.zh-CN", "sans-serif");
 pref("font.size.variable.zh-CN", 16);
-pref("font.size.fixed.zh-CN", 16);
+pref("font.size.fixed.zh-CN", 14);
 
 pref("font.default.zh-TW", "sans-serif");
 pref("font.size.variable.zh-TW", 16);
-pref("font.size.fixed.zh-TW", 16);
+pref("font.size.fixed.zh-TW", 14);
 
 pref("font.default.zh-HK", "sans-serif");
 pref("font.size.variable.zh-HK", 16);
-pref("font.size.fixed.zh-HK", 16);
+pref("font.size.fixed.zh-HK", 14);
 
 // below a certian pixel size outline scaled fonts produce poor results
 pref("font.scale.outline.min",      6);
