--- mozilla/toolkit/mozapps/installer/packager.py.orig	2013-11-13 01:14:17.000000000 +0100
+++ mozilla/toolkit/mozapps/installer/packager.py	2013-11-24 14:15:29.769622782 +0100
@@ -144,7 +144,8 @@
                             '-e', 'precompile_startupcache("resource://%s/");'
                                   % resource],
                            extra_linker_path=gre_path,
                            extra_env={'MOZ_STARTUP_CACHE': cache,
-                                      'PATH': env['PATH']}):
+                                      'PATH': env['PATH'],
+                                      'LD_PRELOAD': os.path.join(gre_path, 'libxul.so')}):
             errors.fatal('Error while running startup cache precompilation')
             return
         from mozpack.mozjar import JarReader
